modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -142,20 +155,167 @@ MultiSmoothSuperellipsoidIC::computeSuperellipsoidCenters()
       }
 
       if (i == 0) rr = _range.norm();
+
+      //Check each new center candidate against all accepted ellisoids unless this is the first candidate
+      if (_prevent_overlap == 1 && rr > _bubspac[_gk] && _centers.size() > 0)
+      {
+        _CentNum = _centers.size();
+        Real status = overlapCheck(newcenter, _as[_CentNum], _bs[_CentNum], _cs[_CentNum], _ns[_CentNum]);
+        //If candidate center fails overlapCheck ensure that rr is smaller than _bubspac to continue loop
+        if (status == 0)
+          rr = _bubspac[_gk] - 1;
+      }
     }
 
     if (num_tries == _numtries)
       mooseError("Too many tries in MultiSuperellipsoidEllipseIC");
 
-    _centers[i] = newcenter;
+    _centers.push_back(newcenter);
   }
 }
 
 void
 MultiSmoothSuperellipsoidIC::computeSuperellipsoidExponents()
 {
-  _ns.resize(_numbub);
+  for (unsigned int el = 0; el < _numbub[_gk]; ++el)
+    _ns.push_back(_exponent[_gk]);
+}
+
+Real
+MultiSmoothSuperellipsoidIC::overlapCheck(const Point & newcenter, const Real & nc_as, const Real & nc_bs, const Real & nc_cs, const Real & nc_ns)
+{
+  Point l_p = newcenter;
+  Real a = nc_as;
+  Real b = nc_bs;
+  Real c = nc_cs;
+  Real n = nc_ns;
+/*
+  Check for overlap between candidate center and accepted centers
+*/
+  //Compute the distance between the current point and the center
+  for (unsigned int el = 0; el < _centers.size(); ++el)
+  {
+    Real dist = _mesh.minPeriodicDistance(_var.number(), _centers[el], l_p);
+    //When dist is 0 we are exactly at the center of the superellipsoid so return _invalue
+    //Handle this case independently because we cannot calculate polar angles at this point
+    if (dist == 0.0) return 0;
+
+    //Compute the distance r from the center of the superellipsoid to its outside edge
+    //along the vector from the center to the current point
+    //This uses the equation for a superellipse in polar coordinates and substitutes
+    //distances for sin, cos functions
+    Point dist_vec = _mesh.minPeriodicVector(_var.number(), _centers[el], l_p);
+
+    //First calculate rmn = r^(-n), replacing sin, cos functions with distances
+    Real rmn = (std::pow(std::abs(dist_vec(0) / dist / a), n)
+              + std::pow(std::abs(dist_vec(1) / dist / b), n)
+              + std::pow(std::abs(dist_vec(2) / dist / c), n) );
+    //Then calculate r1 from rmn
+    Real r1 = std::pow(rmn, (-1.0/n));
+
+    dist_vec = _mesh.minPeriodicVector(_var.number(), l_p, _centers[el]);
 
-  for (unsigned int el = 0; el < _ns.size(); ++el)
-    _ns[el] = _exponent;
+    //First calculate rmn = r^(-n), replacing sin, cos functions with distances
+    rmn = (std::pow(std::abs(dist_vec(0) / dist / _as[el]), _ns[el])
+              + std::pow(std::abs(dist_vec(1) / dist / _bs[el]), _ns[el])
+              + std::pow(std::abs(dist_vec(2) / dist / _cs[el]), _ns[el]) );
+    //Then calculate r2 from rmn
+    Real r2 = std::pow(rmn, (-1.0/_ns[el]));
+
+    if (dist < r1 + r2)
+      return 0;
```

Comment:
Just a minor point, but I prefer if a decimal point is specified here (`0.0`) to make it more clear that this function is returning a floating point value and not an integer.